pkg/object/meshcontroller/ingresscontroller/ingresscontroller.go

```diff
@@ -42,10 +40,12 @@ type (
 		superSpec *supervisor.Spec
 		spec      *spec.Admin
 
-		service   *service.Service
-		pipelines map[string]*httppipeline.HTTPPipeline
-		httpSvr   *httpserver.HTTPServer
-		informer  informer.Informer
+		tc            *trafficcontroller.TrafficController
+		namespace     string
+		httpServer    *supervisor.ObjectEntity
+		httpPipelines map[string]*supervisor.ObjectEntity
```

Comment:
seems `httpServer` and `httpPipelines` are redundant as `tc` already has all the necessary information.

Reply:
According to the original implementation, it's hard to eliminate them at once. They must be deleted at `Close` and so on. I will clean all codes of MeshController in the next contribution.